src/odemis/acq/stitching/_tiledacq.py

```diff
@@ -618,6 +641,81 @@ def _acquireTiles(self):
             i += 1
         return da_list
 
+    def _get_z_on_focus_plane(self, x, y):
+        if not self._focus_plane:
+            c, normal = linalg.fit_plane_lstsq(self._focus_points)
+            self._focus_plane["c"] = c
+            self._focus_plane["normal"] = normal
+        point_on_plane = (0, 0, self._focus_plane["c"])  # where the plane intersects with the z-axis
+        z = linalg.get_z_pos_on_plane(x, y, point_on_plane, self._focus_plane["normal"])
+        return z
+
+    def _get_triangulated_focus_point(self, x, y):
+        """Triangulate focus points and get the z position of the xy-point in the corresponding focus-triangle."""
+        tri = Delaunay(self._focus_points[:, :2])  # triangulate focus points
+
+        point_in_triangle = False
+        # Check in 2D if the point is in one of the triangles from the focus points
+        for simplex in tri.simplices:
+            triangle = [self._focus_points[:, :2][simplex[0]],
+                        self._focus_points[:, :2][simplex[1]],
+                        self._focus_points[:, :2][simplex[2]]]
+            point_in_triangle = point_in_polygon([x, y], polygon=triangle)
+            if point_in_triangle:
+                break
+
+        if point_in_triangle is False:
+            logging.debug("Acquiring tile outside of focused area, will use plane fitting to find the focus.")
+            # If the point is not in one of the triangles fit a plane through all
+            # focus points and base the z position on that plane fit.
+            z = self._get_z_on_focus_plane(x, y)
```

Comment:
Sounds good. The only way I can think of doing this is by calculating the three nearest neighbors and using those as a triangle. But I have the feeling that I'm thinking too complex and there must be a simpler way. @K4rishma or @pieleric do you have a suggestion?

Reply:
Though I can find another simple way. I have another approach in mind. Maybe we can find the location of 6 focus points at the boundary. I think boundary focus points can be calculated from the center of the tiles. We can then omit plane_fit as for any given coordinates, there will be a triangle in place.

Reply:
I'm not sure that finding the 3 closest neighbours would even work well in all cases, because I have a feeling you could end up with 3 points (almost) in line. I think you'd need to look for the 2 closest neighbours, and then, based on these 2 points, find as 3rd point the point closest from both.